src/common/com/intellij/plugins/haxe/ide/HaxeLookupElement.java

```diff
@@ -107,6 +112,28 @@ public void renderElement(LookupElementPresentation presentation) {
     }
   }
 
+  @Override
+  public void handleInsert(InsertionContext context) {
+    HaxeMemberModel memberModel = HaxeMemberModel.fromPsi(myComponentName);
+    boolean hasParams = false;
+    boolean isMethod = false;
+    if (memberModel != null) {
+      if (memberModel instanceof HaxeMethodModel) {
+        isMethod = true;
+        HaxeMethodModel methodModel = (HaxeMethodModel)memberModel;
+        hasParams = !methodModel.getParametersWithContext(this.context).isEmpty();
+      }
+    }
+
+    if (isMethod) {
+      final LookupElement[] allItems = context.getElements();
+      final boolean overloadsMatter = allItems.length == 1 && getUserData(FORCE_SHOW_SIGNATURE_ATTR) == null;
+      JavaCompletionUtil.insertParentheses(context, this, overloadsMatter, hasParams);
+    }
+  }
+  // Workaround for IDEA 14. There must be JavaCompletionUtil.FORCE_SHOW_SIGNATURE_ATTR
+  private static final Key<Boolean> FORCE_SHOW_SIGNATURE_ATTR = Key.create("forceShowSignature");
```

Comment:
You're right.  It's getting late.  

There's already a class called FieldWrapper to do this.  You can see an example of its use in HaxeHierarchyNodeDescriptor.

I'm off to bed.  I'll check this in the morning.

And -- as always -- thanks for the contributions.